packages/bot/src/bot.ts

```diff
@@ -44,9 +44,7 @@ export function createBot(options: CreateBotOptions): Bot {
       // RUN DISPATCH CHECK
       await bot.events.dispatchRequirements?.(data, shard.id)
       bot.events[
-        data.t.toLowerCase().replace(/_([a-z])/g, function (g) {
-          return g[1].toUpperCase()
-        }) as keyof EventHandlers
+        data.t as keyof EventHandlers
```

Comment:
This entire thing from line 46 should be `bot.handlers[data.t]?.();` as the handlers will be the ones to call `bot.events....`